src/lucky_record/errors.cr

```diff
@@ -25,4 +25,11 @@ module LuckyRecord
       super "Invalid #{form_name}. Could not save #{@form_object}"
     end
   end
+
+  # Raised when an unimplemented or deprecated query is made.
+  class UnsupportedQueryError < LuckyRecordError
```

Comment:
I like this description and error. I think it is generic, but the message is specific. So maybe it should accept a message so it can be used for other unsupported queries in the future:

```crystal
def initialize(message : String)
  raise message
end
```